sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+/**
+ * Annotation given to all methods that are generated by AutoRest. This annotation is intended to be used by
+ * the code generation tool only to identify methods that are generated. The purpose of this annotation is to find
+ * and replace all methods in a class that are generated. Methods not annotated with this annotation will not be
+ * updated when code is regenerated.
+ * <p>This annotation is expected to be used in classes that are annotated with {@link ServiceClient} only.</p>
```

Comment:
Do we want to make this more generic and allow it to be used on any generated model?

Reply:
There's no usecase for using this annotation in any other model today but if we find a need to do that, we can update the documentation.